### ISSUE TITLE
Phase out HistoryTrackerInterace

### ISSUE THREAD

**Issue by john-science** [terrapower/armi#792]
It appears @youngmit already did the bulk of the work to move the _functionality_ people want from the `HistoryTrackerInterface` into the Database:  https://github.com/terrapower/armi/commit/38d9febdbec7ab8a67dd9b8e50780e11ea127022

This seem preferable, as the data will be more accessible to anyone with a DB. Also, it reduces the complexity of having a whole `Interface`.  On the surface, it seems like this is an easy change to make.  The (apparent) steps to take now are:

1. Ensure that all the important functionality/data from the `HistoryTrackerInterface` is now in the DB.
2. Find all downstream projects that use the `HistoryTrackerInterface`, and switch them to using the DB.
    - Right now I believe there is an equilibrium plugin that uses the HTI.

**Comment by ntouran** [comment]
Yeah so history tracker was originally made before we could read a case's own database during a run for historical info. It saved a subset of state data to ram that it could very quickly bring back as needed. Now that db3 can read itself during a run, it's largely not needed. 

The internal equilibrium capability has lots of fanciness that maps an equilibrium cycle's shuffling into full multi-cycle time-dependent histories that must be maintained internally, but that can be done somewhere outside of the history interface.

**Comment by john-science** [comment]
> The internal equilibrium capability has lots of fanciness...

Heck, I could copy/paste all the code they need into their project, too.  If they need more than the "new" DB3 can provide.

**Comment by john-science** [comment]
@ntouran Hey Nick.

@onufer was saying he remembers talking to you about this ticket, and he thought we shouldn't remove this interface, but pull a small set of key features out of it and add it to the `Datavase3` or `DatabaseInterface`.

I remember opening this ticket after talking to you, but it was obviously a _while_ ago, so I thought I'd just check base with you again.

> What were your expectations for the work here?

Thanks!

**Comment by john-science** [comment]
Okay, I found several downstream projects that use this `Interface` as-is.

And the core logic of the History interface is definitely interface-centric: it's based around an interactBOL, interactBOC, interactEOL, etc.

And a lot of it is specific to reactors with assemblies. 

So I guess I don't want to put all that into the Database3 or DatabaseInterface class.  I'll do a little more research before trying to phase it out.
